xrpl/asyncio/wallet/wallet_generation.py

```diff
@@ -143,9 +148,14 @@ async def _check_wallet_balance(address: str, client: Client) -> int:
         raise
 
 
-async def _request_funding(url: str, address: str) -> None:
+async def _request_funding(
+    url: str, address: str, usage_context: Optional[str] = None
+) -> None:
     async with httpx.AsyncClient() as http_client:
-        response = await http_client.post(url=url, json={"destination": address})
+        json_body = {"destination": address, "userAgent": "xrpl-py"}
```

Comment:
I think user agent should also be an input parameter that defaults to xrpl-py but the user can change to be None if they want